.github/workflows/changelog.py

```diff
@@ -0,0 +1,202 @@
+
+Adds a line into the CHANGELOG.md:
+If a PR title starts with "New module: ", adds a line under the ""### New modules" section.
+If a PR starts with a name of an existing module, adds a line under "### Module updates".
+Everything else will go under "MultiQC updates" in the changelog, unless "(chore)" or "(docs)" is appended to the title.
+
+Other assumptions:
+- CHANGELOG.md has a running section for an ongoing "dev" version (i.e. titled "## MultiQC vX.Ydev").
+- Under that section, there are sections "### MultiQC updates", "### New modules" and "### Module updates".
+- For module meta info, checks the file multiqc/modules/<module_name>/<module_name>.py.
+"""
+
+import os
+import re
+import sys
+from pathlib import Path
+
+CHANGELOG_PATH = "../../CHANGELOG.md"
+REPO_URL = "https://github.com/ewels/MultiQC"
+
+# Get the PR title and description from the environment
+pr_title = os.environ["PR_TITLE"]
+pr_number = os.environ["PR_NUMBER"]
+
+# Trim the PR number automatically appended when GitHub squashes commits, e.g. "Module: Updated (#2026)".
+if not pr_title.endswith(f" (#{pr_number})"):
+    print(
+        f"Note: the PR title '{pr_title}' doesn't end with PR number {pr_number}, which "
+        f"likely means it was pushed directly into master, or wasn't squashed",
+        file=sys.stderr,
+    )
+    sys.exit(0)
+pr_title = pr_title.removesuffix(f" (#{pr_number})")
+
+
+# If "(chore)" or "(docs)" is appended to the PR title, it indicates that we don't want to log this change.
+if pr_title.endswith("(chore)") or pr_title.endswith("(docs)"):
+    print("Skipping logging this change as it's a chore or docs update")
+    sys.exit(0)
+
+
+def find_module_info(module_name):
+    """
+    Helper function to load module meta info. With current setup, can't really just import
+    the module and call `mod.info`, as the module does the heavy work on initialization.
+    But that's good - we avoid installing and importing MultiQC here, and the action runs faster.
+    """
+    module_name = module_name.lower()
+    modules_dir = Path("../../multiqc/modules")
+    py_path = None
+    for dir_name in os.listdir(modules_dir):
+        if dir_name.lower() == module_name:
+            module_dir = modules_dir / dir_name
+            py_path = module_dir / f"{dir_name}.py"
+            if not py_path.exists():
+                print(f"Folder for {module_name} exists, but doesn't have a {py_path} file", file=sys.stderr)
+                sys.exit(1)
+            break
+
+    if not py_path:  # Module not found
+        return None
+    with py_path.open("r") as f:
+        contents = f.read()
+    if not (m := re.search(r'name="([^"]+)"', contents)):
+        return None
+    name = m.group(1)
+    if not (m := re.search(r'href="([^"]+)"', contents)):
+        return None
+    url = m.group(1)
+    if not (m := re.search(r'info="([^"]+)"', contents)):
+        if not (m := re.search(r'info="""([^"]+)"""', contents)):
```

Comment:
Will this match strings including newlines?

Reply:
Yep! Tested :)